Eigen/src/Core/products/GeneralBlockPanelKernel.h

```diff
@@ -40,7 +40,14 @@ const std::ptrdiff_t defaultL3CacheSize = 512*1024;
 struct CacheSizes {
   CacheSizes(): m_l1(-1),m_l2(-1),m_l3(-1) {
     int l1CacheSize, l2CacheSize, l3CacheSize;
+
+#if !defined(EIGEN_CUDA_ARCH)
     queryCacheSizes(l1CacheSize, l2CacheSize, l3CacheSize);
+#else
```

Comment:
@vkhristenko would it make sense to reuse the cache definition from `Eigen/src/Core/util/Memory.h` ?

Reply:
this is not a complete reuse. the logic i had is

- L1 -> 16KB ( 48 KB shared ) which is accurate 
- L2 -> query device was previously on the order of 1MB (on cmg-gpu cards now is 2MB)
- L3, there is no such thing as L3 really (as far as i understand the arch of nvidia cards)...

any suggestions are more than welcome here, this was the logic and seemed to not cause any trouble

Reply:
OK.

For L1, we could make the hardcoded value depend on the device model reported by `__CUDA_ARCH__`.

For L2, do you mean that `cudaDeviceGetAttribute(...)` did not return the right results ?

For L3, yes, as far as I know there is no L3 cache.

Reply:
For L1, that would indeed be more appropriate

For L2, I just ran `queryDevice` to check what L2 was...